src/compiler/checker.ts

```diff
@@ -27318,8 +27307,11 @@ namespace ts {
                 // analysis because variables may have transient types in indeterminable states. Moving flowLoopStart
                 // to the top of the stack ensures all transient types are computed from a known point.
                 const saveFlowLoopStart = flowLoopStart;
+                const saveFlowTypeCache = flowTypeCache;
                 flowLoopStart = flowLoopCount;
+                flowTypeCache = undefined;

```

Comment:
Since we do it in two places, can we capture this "execute in new flow cache context" pattern into it's own helper function?

Reply:
I think it is fine as is.